### PR TITLE
feat(cli): add library flag for query

### PR DESCRIPTION
**Pull request by hazedav**
## Summary
Add library flag for `lacework query`
```
lacework query list --library
lacework query show LW_Custom_Host_Activity_CodecovCurl --library
lacework query create --library LW_Custom_Host_Activity_CodecovCurl 
lacework query update --library LW_Custom_Host_Activity_CodecovCurl 
lacework query run LW_Custom_Host_Activity_CodecovCurl --library
lacework query validate --library LW_Custom_Host_Activity_CodecovCurl 
```

## How did you test this change?
Automated testing added
Manual testing completed

## Issue
https://lacework.atlassian.net/browse/ALLY-839

### CODE REVIEW

File: api/lql_test.go [hunk 1]
```diff
@@ -24,9 +24,11 @@ import (
 	"net/http"
 	"testing"
 
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+
```
Comment (afiune) [inline api/lql_test.go:29]: I love this! Thank you

File: cli/cmd/lql_library.go [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+		out = append(out, []string{id})
+	}
+	return
+}
+
+func listQueryLibrary(_ *cobra.Command, args []string) error {
+	cli.Log.Debugw("listing queries from library")
+
+	cli.StartProgress(" Retrieving queries...")
+	lcl, err := LoadLCL(*cli.LwComponents)
```
Comment (afiune) [inline cli/cmd/lql_library.go:60]: why this is not like
```suggestion
	lcl, err := cli.LwComponent.LoadLCL()
```

**Review by afiune — APPROVED** [review]
One comment but please, start merging things since I have to see the entire result. 👍🏽

File: cli/cmd/lql_library.go (hunk 2 above)
Comment (hazedav) [reply cli/cmd/lql_library.go:60]: `LwComponents` is a pointer to `lwcomponent.State`:
```
LwComponents *lwcomponent.State
```

We don't want `LoadLCL` to be a method of `lwcomponent.State`, but we could make this a method of `cliState`...

Thoughts?

Comment (hazedav) [reply cli/cmd/lql_library.go:60]: Going to merge this down, but let's continue this discussion...

**Pull request merged**